### ISSUE TITLE
Config.HTTPS and Config.Secrets are too limited

### ISSUE THREAD

**Issue by halostatue** [nccgroup/sobelow#7]
Config.HTTPS and Config.Secrets only look in `config/prod.exs` (HTTPS and Secrets) and `config/prod.secret.exs` (Secrets). We have three *public-facing* environments (`dit`, `staging`, and `production`; it’s called `production` for compatibility with our Rails deploy configurations).

As such, I’m getting a report on `Config.HTTPS` even though all three of my public-facing environments have both HTTPS and HSTS configured. I recommend that `Config.HTTPS` and `Config.Secrets` have two major changes:

1. Change to an all-files scan (`config/(.+)(?:\.secret)?\.exs`) instead of specific files. In reporting, report on each file that has the problem unless the captured part is `dev` or `test`.

2. Allow for configuration of included or excluded files. This second is probably bigger, because other than code decorators, I’m not seeing any configuration options.

**Comment by GriffinMB** [comment]
I agree that the scope is too narrow, especially for Secrets. I'll expand the Secrets checks to scan all config files except dev/test. I'll also make some changes to the HTTPS/HSTS checks -- I don't think I'll expand the scope in the same way, because that will just lead to more false positives. But I've got some ideas on how to improve them. 

As for configuration for excluding files, I'm working on adding config functionality and will make sure that's included. Since the decorator-like syntax won't work for things like config and dependency checks, I think being able to exclude files from the scan will be important.

**Comment by GriffinMB** [comment]
I pushed 0.3.8 earlier this week, which included some HTTPS/HSTS/Secrets improvements. A brief rundown:

* All configs (except dev and test) in the config directory are scanned now.
* Separates HTTPS/HSTS findings. Any time HTTPS is configured, it will expect to find HSTS configuration.
* It's still only checking for HTTPS at prod.exs, but now won't flag it as a finding if the file doesn't exist. 
* Pseudo fuzzy finding for secrets, so it will now catch things like AWS_SECRET in addition to secret_key_base.

I'll close this once I've implemented configuration, but it doesn't look like that will happen this week. I'm also going to continue thinking about the HTTPS finding.
